weave_demo.ipynb

```diff
@@ -0,0 +1,1199 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "id": "ddfd3e45-691d-4465-a21a-748a3dd843fa",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>uuid</th>\n",
+       "      <th>upload_time</th>\n",
+       "      <th>parent_uuids</th>\n",
+       "      <th>basket_type</th>\n",
+       "      <th>label</th>\n",
+       "      <th>weave_version</th>\n",
+       "      <th>address</th>\n",
+       "      <th>storage_type</th>\n",
+       "      <th>generation_level</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>acc530466eb811eeb8b60242ac1f000a</td>\n",
+       "      <td>2023-10-19 19:49:54.672040+00:00</td>\n",
+       "      <td>[]</td>\n",
+       "      <td>test-1</td>\n",
+       "      <td></td>\n",
+       "      <td>1.1.0</td>\n",
+       "      <td>weave-demo-pantry/test-1/acc530466eb811eeb8b60...</td>\n",
+       "      <td>LocalFileSystem</td>\n",
+       "      <td>1</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                               uuid                      upload_time  \\\n",
+       "0  acc530466eb811eeb8b60242ac1f000a 2023-10-19 19:49:54.672040+00:00   \n",
+       "\n",
+       "  parent_uuids basket_type label weave_version  \\\n",
+       "0           []      test-1               1.1.0   \n",
+       "\n",
+       "                                             address     storage_type  \\\n",
+       "0  weave-demo-pantry/test-1/acc530466eb811eeb8b60...  LocalFileSystem   \n",
+       "\n",
+       "   generation_level  \n",
+       "0                 1  "
+      ]
+     },
+     "execution_count": 18,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pantry1.index.get_parents(pantry1_df['uuid'][1])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "64e385eb-9c4a-4077-aa57-bf7a73a77fa9",
+   "metadata": {},
+   "source": [
+    "## Generating an index using the SQLite backend\n",
+    "When creating a pantry using SQLite we can generate indexes corresponding to rows of basket data. For this pantry we will using SQLite for the index backend. An Index is an object or file that tracks the baskets in a pantry."
```

Comment:
Careful on the phrasing here. SQLite does not generate indices, the index is represented as a SQLite object. The generate_index function just makes the index up to date with the contents of the pantry.

It may be beneficial here to explain why we have the abstracted Index backend. Pandas is nice because it is easy to use and familiar. But it is SLOW when the pantry starts getting big. Weave supports SQLite backend for larger pantries, and users can implement their own backend according to their needs.